src/resource/nonRdfData.test.ts

```diff
@@ -38,10 +38,57 @@ import {
   saveFileInContainer,
   overwriteFile,
   getFileWithAcl,
+  flattenHeaders,
 } from "./nonRdfData";
 import { Headers, Response } from "cross-fetch";
 import { WithResourceInfo } from "../interfaces";
 
+describe("flattenHeaders", () => {
+  it("returns an empty object for undefined headers", () => {
+    expect(flattenHeaders(undefined)).toEqual({});
+  });
+
+  it("returns well-formed headers as-is", () => {
+    const headers: Record<string, string> = {
+      test: "value",
+    };
+    expect(flattenHeaders(headers)).toEqual(headers);
+  });
+
+  it("transforms an incoming Headers object into a flat headers structure", () => {
+    const myHeaders = new Headers();
+    myHeaders.append("accept", "application/json");
+    myHeaders.append("content-type", "text/turtle");
+    // The following needs to be ignored because `node-fetch::Headers` and
+    // `lib.dom.d.ts::Headers` don't align. It doesn't break the way we
+    // use them currently, but it's something that must be cleaned up
+    // at some point.
+    // eslint-disable-next-line
+    // @ts-ignore
```

Comment:
Can't we just fix that by having `flattenHeaders` accept both cross-fetch's and the regular `Headers`? It should flatten them in the same, compatible way, so it sounds like that should work fine?

Reply:
And if not, we'd probably better use `@ts-expect-error`, so that we get notified when we can remove this: https://devblogs.microsoft.com/typescript/announcing-typescript-3-9-beta/#ts-expect-error-comments

Reply:
You're right, that's not needed anymore. Removed in 1934686aab5c9f96dfdfbec3d623fe336e6c9ea4.